### PR TITLE
Add then to if statements using new control syntax

### PR DESCRIPTION
**Pull request by kpbochenek**
Hi,

I am working on scalameta parser and trying to reach a stage where we can finally parse whole dotty codebase and enable continuous integration, I am unsure if this syntax `if condition <indented block>` without then is supported but it is very hard to parse right now.

- as adding then doesn't change anything this should make no difference for you, correct?
- is this syntax officially supported or an artifact of previous experiments?

### CODE REVIEW

**Review by odersky — APPROVED** [review]
The `then` should definitely be added. I was on the fence for a while whether we want to support it it the end find it is better not to. A possible exception (for now) is if the condition is in parentheses, i.e. old syntax. Then we still support indentation in order not to entangle if-then-else syntax style with optional braces. But that's not completely settled either.

File: compiler/src/dotty/tools/dotc/typer/Inliner.scala [hunk 1]
```diff
@@ -72,7 +72,8 @@ object Inliner {
   def inlineCall(tree: Tree)(using Context): Tree = {
     val startId = ctx.source.nextId
 
-    if tree.symbol.denot != SymDenotations.NoDenotation && tree.symbol.owner.companionModule == defn.CompiletimeTestingPackageObject
+    if tree.symbol.denot != SymDenotations.NoDenotation
+        && tree.symbol.owner.companionModule == defn.CompiletimeTestingPackageObject then
```
Comment (nicolasstucki) [inline compiler/src/dotty/tools/dotc/typer/Inliner.scala:76]: Maybe the folowing would be slightly easier to read
```suggestion
       && tree.symbol.owner.companionModule == defn.CompiletimeTestingPackageObject
    then
```

Comment (kpbochenek) [reply compiler/src/dotty/tools/dotc/typer/Inliner.scala:76]: yeah that would be more readable, but you merged it already so I don't know what now, create single PR with this single change? ❓

Comment (kpbochenek) [reply compiler/src/dotty/tools/dotc/typer/Inliner.scala:76]: will do another PR
